dbms/src/Common/FailPoint.cpp

```diff
@@ -139,8 +140,9 @@ namespace DB
     M(random_pipeline_model_event_schedule_failpoint)   \
     M(random_pipeline_model_event_finish_failpoint)     \
     M(random_pipeline_model_operator_run_failpoint)     \
-    M(random_pipeline_model_cancel_failpoint)
-
+    M(random_pipeline_model_cancel_failpoint)           \
+    M(random_join_spill_to_disk_failpoint)              \
+    M(random_join_restore_from_disk_failpoint)
```

Comment:
how about replacing by `random_spill_to_disk_failpoint` and `random_restore_from_disk_failpoint`. and injecting fail point in `Spiller::spill` and `Spiller::restore`.
Then not just join, but all spill operators are injected with fail points.

Reply:
> how about replacing by `random_spill_to_disk_failpoint` and `random_restore_from_disk_failpoint`. and injecting fail point in `Spiller::spill` and `Spiller::restore`. Then not just join, but all spill operators are injected with fail points.

The purpose of this failPoint is to test the join executor， if injecting fail point in Spiller it is not possible to determine whether the problem is in the join or another executor when the failPoint test fails.

Reply:
random fail point test is the chaos test for random fault injection.
IMO, `random_spill_to_disk_failpoint` and `random_restore_from_disk_failpoint` is more suitable for random fail point test.
Not only join, but also other errors in the falling spill operator should be detected.

Finally, in case of a crash, you can determine from the stack log if it was caused by a join.
If it's a hang, then there's no way to tell if it's a join issue even if the fail point is only injected against a join.

Reply:
done.